src/classes/AFFL_MultiRecordType_TDTM.cls

```diff
@@ -94,7 +94,7 @@ public class AFFL_MultiRecordType_TDTM extends TDTM_Runnable {
 					}
 
 					// AFTER UPDATE
-					if (triggerAction == TDTM_Runnable.Action.AfterUpdate && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_MultiRecordType_TDTM_After_Update)) {
+					if (triggerAction == TDTM_Runnable.Action.AfterUpdate && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_MultiRecordType_TDTM_After_Update) && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_MultiRecordType_TDTM_afflMadePrimary)) {
 						Affiliation__c afflOld = (Affiliation__c)oldlist[i];
```

Comment:
I'm ok using `TDTM_ProcessControl.getRecursionFlag` since it's done all over this file. I would however ask for 2 things:

1. Let's break line 97 into multiple lines so there is no need to scroll so much to the right. We can do that on every &&. Since we are on it, let's also do it for line 88 (this is more consistent with the rest of the file). See screenshot below.

2. Let's leave a comment clarifying why we are calling `TDTM_ProcessControl.getRecursionFlag`. If this could be done in all places in the file where this method is called that'd be great. :-) Frankly, I still don't fully understand how it works.

![Screen Shot 2019-10-31 at 4 38 19 PM](https://user-images.githubusercontent.com/1894238/67988266-746b5680-fbfd-11e9-8d39-c91c1234e58b.png)

Reply:
Thanks for the feedback Carlos, I will make the changes. Meanwhile, I made a quip doc with root cause not sure if this helps https://salesforce.quip.com/WMhTACdopneZ.

Reply:
@ceiroa